operator/api/vectorized/v1alpha1/cluster_types.go

```diff
@@ -1137,8 +1215,13 @@ func (r *Cluster) SchemaRegistryAPIURL() string {
 	// Prefer to use internal URL
 	// But if it is externally available with TLS, we cannot call internal URL without TLS and the TLS certs are signed for external URL
 	host := r.Status.Nodes.SchemaRegistry.Internal
-	if r.IsSchemaRegistryExternallyAvailable() && r.IsSchemaRegistryTLSEnabled() {
-		host = r.Status.Nodes.SchemaRegistry.External
+	allSchemaRegistry := r.AllSchemaRegistryListeners()
```

Comment:
This function appears to have exactly one consumer and your change has made the expected behavior very ambiguous. I think you'd be better served by removing this and related functions and instead replace it with a single function that returns a `config.Schema` struct suitable for consumption by Console.

From looking at the usages, it seems like you could wipe out a good chunk of these methods. In the cases where there are multiple consumers, I think you'll be better served by either inverting the behavior or by making these methods accept a `name` parameter.